docs/manual/src/internals/ffi_converter_traits.md

```diff
@@ -0,0 +1,68 @@
+## Rust FFI conversion traits
+
+UniFFI leverages a set of FFI converter traits to implement [lifting and lowering](./lifting_and_lowering.md) on the Rust side.
+Each trait handles a single step in the lifting/lowering process (e.g. lifting an argument, lowering a return, etc.).
+We implement the trait for each type used in the exported API then use that trait in the codegen.
+
+For example, `uniffi::Lift` is used to lift values.
+To handle a function like `fn print(msg: String)`, the generated code will use:
+
+- `<String as Lift<crate::UniFfiTag>>::FfiType` when it needs to specify the FFI type (`RustBuffer`).
+- `<String as Lift<crate::UniFfiTag>>::try_lift()` when it needs to lift an argument value.
+
+Using a trait for this is important for proc-macros, which only see Rust tokens and don't know the surrounding context.
+For example, if macros always used `RustBuffer` as the FFI type whenever it sees "String", then that would fail if users created a type alias like `type String = u8`.
```

Comment:
`type String = u8` seems like an odd example - would using something like `Guid` or even `Url` be better? It makes `String` seems less unusual.